framework/src/media/FileOutputDataSource.cpp

```diff
@@ -91,11 +91,17 @@ bool FileOutputDataSource::isPrepare()
 ssize_t FileOutputDataSource::write(unsigned char* buf, size_t size)
 {
 	if (!buf) {
-		meddbg("buf is nullptr, hence return 0\n");
-		return (size_t)0;
+		meddbg("buf is nullptr, hence return ERROR\n");
+		return EOF;
 	}
-
-	return fwrite(buf, sizeof(unsigned char), size, mFp);
+	
+	size_t ret;
+	ret = fwrite(buf, sizeof(unsigned char), size, mFp);
```

Comment:
mixing upper and lower case for a variable name "mFp"  may not be recommended

Reply:
that is common style of c++